tests/test_agent_modules/test_tool_factory_langchain.py

```diff
@@ -0,0 +1,119 @@
+"""
+Unit tests for ToolFactory LangChain integration.
+
+Tests the LangChain tool conversion functionality with REAL LangChain tools.
+"""
+
+from unittest.mock import patch
+
+import pytest
+from agents import FunctionTool
+
+from agency_swarm.tools.tool_factory import ToolFactory
+
+
+class TestLangchainIntegration:
+    """Test REAL LangChain tool integration."""
+
+    def test_missing_langchain_import(self):
+        """Test error when langchain_community is not available."""
+
+        # Create a simple dummy tool class
+        class DummyTool:
+            def run(self):
+                return "dummy"
+
+        dummy_tool = DummyTool()
+
+        # Mock import to simulate ImportError
```

Comment:
Capture the original __import__ before patching in the LangChain tool factory test to avoid recursive mock calls when simulating missing dependencies

```suggestion
        # Mock import to simulate ImportError
        original_import = builtins.__import__
```